locks: make spack's locks work on windows

loulawrence opened the pull request:
Spack on Windows Draft

loulawrence committed:
Windows pywin32 locking

tgamblin commented on .gitignore:
```diff
@@ -38,3 +38,11 @@ lib/spack/spack/test/.cache
+/var/spack/repos/builtin/packages/pythia6/out/build/x64-Debug (default)/CMakeFiles
+/var/spack/repos/builtin/packages/tinyxml/out/build/x64-Debug (default)/CMakeFiles
+/var/spack/repos/builtin/packages/pythia6/out/build/x64-Debug (default)
+/var/spack/repos/builtin/packages/tinyxml/out/build/x64-Debug (default)
```
@loulawrence these seem like they should not be in the mian Spack repo

BetsyMcPhail committed:
Locking ranges is not currently supported on Windows

Disable all lock tests that require ranges or that are not currently
passing on Windows.

BetsyMcPhail commented:
Making progress on running the lock tests (``spack unit-test lib/spack/spack/test/llnl/util/lock.py``) on Windows. In the latest commit, 6 tests fail and 1 hangs, they are all disabled (e.g. start with ``def _test...``) and are marked with "TODO"

BetsyMcPhail committed:
os.rename() fails on Windows if file already exists

BetsyMcPhail commented:
With the exception of code coverage, all tests are passing. This PR is ready for review and to be merged onto `spack:features/windows-support`

@zackgalbreath or @benc303 can you PTAL ?
